lib-api/src/main/java/org/ergoplatform/appkit/InputBoxesValidator.scala

```diff
@@ -0,0 +1,109 @@
+package org.ergoplatform.appkit
+
+import org.ergoplatform.wallet.Constants.MaxAssetsPerBox
+import org.ergoplatform.wallet.boxes.BoxSelector._
+import org.ergoplatform.wallet.boxes.DefaultBoxSelector._
+import org.ergoplatform.wallet.boxes.{BoxSelector, ReemissionData}
+import org.ergoplatform.wallet.{AssetUtils, TokensMap}
+import org.ergoplatform.{ErgoBoxAssets, ErgoBoxAssetsHolder}
+import scorex.util.ModifierId
+import sigmastate.utils.Helpers.EitherOps
+
+import scala.collection.mutable
+
+/**
+  * Pass through implementation of the box selector. Unlike DefaultBoxSelector from ergo-wallet,
+  * it does not select input boxes. We do this in appkit ourselves and do not need the selector
+  * to interfere with how we built our transaction. Instead, this selector performs validation
+  * and calculates the necessary change box
+  */
+class InputBoxesValidator extends BoxSelector {
+
+  override def reemissionDataOpt: Option[ReemissionData] = None
+
+  override def select[T <: ErgoBoxAssets](inputBoxes: Iterator[T],
```

Comment:
I can try, but will Scala let me do this change while still overriding the interface? Would not work on Java and Kotlin.

Reply:
Then maybe use some immutable Java collection, especially because immutability is part of the method contract.
I see the iterator comes from Scala collection, so conversion will be necessary anyway. 
Maybe you just use the iterator in a functional way using `foreach` (not imperative hasNext, next() iteration)

Reply:
Using `Iterable` is not possible because the interface defines it as `Iterator`, but using `foreach` is of course possible and better. Interesting that the original code used `hasNext` etc